sample/src/main/java/com/google/modernstorage/sample/mediastore/MediaStoreViewModel.kt

```diff
@@ -21,129 +21,113 @@ import androidx.lifecycle.AndroidViewModel
     private val savedStateHandle: SavedStateHandle
 ) : AndroidViewModel(application) {
     private val httpClient by lazy { OkHttpClient() }
-    private val mediaStore by lazy { MediaStoreClient(application) }
+    private val mediaStore by lazy { MediaStoreRepository(application) }
 
     val canWriteInMediaStore: Boolean
-        get() = canWriteOwnEntriesInMediaStore(getApplication())
+        get() = mediaStore.canWriteOwnEntries()
 
     private val _isLoading = MutableLiveData(false)
     val isLoading: LiveData<Boolean> get() = _isLoading
 
     fun setLoadingStatus(isLoading: Boolean) {
-        _isLoading.value = isLoading
+        _isLoading.postValue(isLoading)
     }
 
-    private val _currentMedia: MutableLiveData<MediaResource?> = MutableLiveData(null)
-    val currentMedia: LiveData<MediaResource?> get() = _currentMedia
-
-    init {
-        savedStateHandle.get<Uri>("currentMediaUri")?.let { uri ->
-            if (canWriteInMediaStore) {
-                viewModelScope.launch {
-                    _currentMedia.value = mediaStore.getResourceByUri(uri)
-                }
-            }
-        }
-    }
+    val currentFile: LiveData<FileResource> = savedStateHandle.getLiveData(CURRENT_MEDIA_KEY)
 
-    fun setCurrentMedia(uri: Uri) {
-        viewModelScope.launch {
-            mediaStore.getResourceByUri(uri)?.let {
-                savedStateHandle.set("currentMediaUri", uri)
-                _currentMedia.value = mediaStore.getResourceByUri(uri)
-            }
-        }
+    suspend fun setCurrentMedia(uri: Uri) {
+        savedStateHandle.set(CURRENT_MEDIA_KEY, mediaStore.getResourceByUri(uri).getOrNull())
     }
 
     val temporaryCameraImageUri: Uri?
-        get() = savedStateHandle.get("temporaryCameraImageUri")
+        get() = savedStateHandle.get(TEMPORARY_CAMERA_IMAGE_URI_KEY)
 
     fun saveTemporaryCameraImageUri(uri: Uri) {
-        savedStateHandle.set("temporaryCameraImageUri", uri)
+        savedStateHandle.set(TEMPORARY_CAMERA_IMAGE_URI_KEY, uri)
     }
 
     fun clearTemporaryCameraImageUri() {
-        savedStateHandle.remove<Uri>("temporaryCameraImageUri")
+        savedStateHandle.remove<Uri>(TEMPORARY_CAMERA_IMAGE_URI_KEY)
     }
 
-    fun saveRandomImageFromInternet(callback: (uri: Uri) -> Unit) {
-        viewModelScope.launch {
-            val request = Request.Builder().url(SampleData.image.random()).build()
-
-            withContext(Dispatchers.IO) {
-                val response = httpClient.newCall(request).execute()
-
-                response.body?.use { responseBody ->
-                    val filename = generateFilename(MediaSource.INTERNET, "jpg")
-                    val imageUri = mediaStore.addImageFromStream(
-                        filename = filename,
-                        inputStream = responseBody.byteStream(),
-                        location = SharedPrimary
-                    )
+    suspend fun saveRandomMediaFromInternet(type: FileType): Result<FileResource> {
```

Comment:
```suggestion
    fun saveRandomMediaFromInternet(type: FileType): Result<FileResource> {
```

This function is not using any coroutine API, so it shouldn't be a suspend function